charts/redis/templates/service.yaml

```
@@ -36,6 +36,7 @@ spec:
       nodePort: {{ $val.nodePort }}
       {{- end }}
       protocol: {{ default "TCP" $val.protocol | quote }}
+      appProtocol: redis
```

I would suggest to wrap the setting to be able to control it via the values file:

```suggestion
      {{- with $val.appProtocol }}
      appProtocol: {{ . }}
      {{- end -}}
```